interview-questions.md

```diff
@@ -8,48 +8,62 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 Researched answer:
 
+Yarn is an open-source project manager that Facebook introduced with the collaboration of Google, Exponent, and Tilde. Yarn allows data sharing between developers to occur faster, more securely, and more reliably. 
+
 3. What is a React component?
 
-Your answer:
+Your answer: 
+A component in React allows the dedication of specific functions to perform specific operations. I like to compare React components to engine components since an engine is made up of a series of electrical/mechanical components that allows the engine to run. Like in React, an engine component, if designed correctly, will allow the engine to run more effectively. Additionally, it may allow the developer/mechanic to troubleshoot and repair specific components more quickly.
 
 Researched answer:
+A React component is a stand-alone function that developers can use to store behavioral code for specific instances. The use of components compartmentalizes the code inside of a web application. This allows the rendering process to render specific features/items and render other components when specific user interactions occur. If I wanted to create a UI web application use a chat box or a like button feature. I could create a specific component for each of the features.
+
 
 4. What is the difference between state values and props in React?
 
-Your answer:
+Your answer: 
+A react state is a special variable that allows a component to perform or behave by storing data. While a react prop is a read-only property that allows a parent component to communicate with its child component. 
 
 Researched answer:
+    React Props: allows communication to flow from parent components to child components. 
+    React States: allows the management of data only within a component.
 
 5. What is the DOM?
 
-Your answer:
+Your answer: Document Object Model is used to track a user's interaction, such as key strikes, cursor movement, or eye movements.
 
 Researched answer:
 
+DOM is like a map that allows the computer to know where each of the elements of a web page are located. Developers can change different parts of the web page and create more dynamic and interactive web pages.  
+
+
 6. STRETCH: Which is the difference between a div tag and a span tag?
 
-Your answer:
+Your answer: I don't know what a span tag is, but if I were to take a wild guess. A span tag <></> is used to encase different types of tags that are used alongside each other within a 'return.' In contrast, <div> tags render a break(space) on a web page. For instance, if I have two paragraphs encased in <p> tags, inserting <div> between both paragraphs produces a separation between the two sections once rendered on a browser. 
 
-Researched answer:
+Researched answer: I needed to be corrected about <span> tags and <div> tags. While both <span> and <div> tags create a container, the <div> can create a container around multiple elements, and the <span> can create a container within each element. 
```

Comment:
The span tag is used to target an inline element while a div is for a block of code.